ci/webhook-receiver.py

```diff
@@ -195,13 +216,28 @@ def init_ci_pipeline():
     out, err = proc2.communicate()
 
     if (out):
-        print("Can't run CI, another CI run in progress")
-        log_access_granted(request_ctx, "CI busy, posting busy msg")
-        os.system("./ci_busy.sh config {} \"{}\" \"{}\" \"Another CI run in progress, please try again in 15 minutes\""
-                  .format(request_ctx['id'], request_ctx['repo'], request_ctx['body']))
+        print("Can't run CI, another CI run in progress, placing in the queue")
+        log_access_granted(request_ctx, "CI busy, placing request in queue")
+        duplicate_req = False
+        busy_msg = "Duplicate request already waiting, ignoring message"
+        # make sure we don't bypass thread safety
+        with queue_lock:
+            for req in ci_request_list:
+                # make sure this is the same PR
+                if req['id'] == request_ctx['id'] and req['repo'] == request_ctx['repo']:
+                    duplicate_req = True
+                    break
+            if not duplicate_req:
+                # don't have this request yet, put in queue
+                ci_request_list.append(request_ctx)
+                busy_msg = "Another CI run in progress, adding request to the end of the list"
+        # ending frees the lock
+        os.system("./ci_busy.sh config {} \"{}\" \"{}\" \"{}\""
```

Comment:
Lets move this up, send this message right after we determine out,err status

Reply:
We can't do that because of the `busy_msg`. We want to tell the user if the request will be ignored as a duplicate